src/node.ts

```diff
@@ -1,15 +1,18 @@
 import formats from '@rdfjs/formats-common'
 import dataset from '@rdfjs/dataset'
+import { DatasetCore } from '@rdfjs/types'
 import * as Alcaeus from './index'
 import 'isomorphic-fetch'
 
-export function create(opts?: Partial<Parameters<typeof Alcaeus['create']>[0]>) {
-    return Alcaeus.create({
+export function create<D extends DatasetCore>(opts: Partial<Omit<Alcaeus.Init<D>, 'fetch' | 'Headers'>> = {}) {
+    const { datasetFactory, ...rest } = opts
+
+    return Alcaeus.create<D>({
         fetch,
         Headers,
         parsers: formats.parsers,
-        datasetFactory: dataset.dataset,
-        ...opts,
+        datasetFactory: datasetFactory || dataset.dataset as any,
```

Comment:
Why do you need `as any`?

Reply:
Because while the outer function is generic, typescript cannot complains it cannot guarantee that `dataset.dataset` actually matches whatever argument you might type (not using inference).

```typescript
const client = create<DatasetExt>()
```

The above is technically correct but obviously wrong since the dataset will not actually implement `DatasetExt`. For the same reason, without `as any` there is an error warning that the type `D` may not match. I don't know how to make it really accurate